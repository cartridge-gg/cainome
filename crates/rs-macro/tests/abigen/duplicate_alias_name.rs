#![no_main]
use cainome_rs_macro::abigen;

abigen!(
    MyStruct,
    r#"[
        {
            "type": "struct",
            "name": "core::integer::u256",
            "members": [
              {
                "name": "low",
                "type": "core::integer::u128"
              },
              {
                "name": "high",
                "type": "core::integer::u128"
              }
            ]
        },
        {
            "type": "struct",
            "name": "core::starknet::eth_address::EthAddress",
            "members": [
              {
                "name": "address",
                "type": "core::felt252"
              }
            ]
        }
    ]"#,
    type_aliases {
        core::integer::u256 as MyStruct;
        core::starknet::eth_address::EthAddress as MyStruct;
    }
);
