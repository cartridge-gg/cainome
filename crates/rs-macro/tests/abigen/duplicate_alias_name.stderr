error: MyStruct duplicate alias name
  --> tests/abigen/duplicate_alias_name.rs:34:9
   |
34 |         core::starknet::eth_address::EthAddress as MyStruct;
   |         ^^^^
